src/item.cpp

```diff
@@ -2881,6 +2881,7 @@ std::string item::tname( unsigned int quantity, bool with_prefix ) const
     } else {
         maintext = label( quantity );
     }
+    maintext = colorize( maintext, color_in_inventory() );
```

Comment:
I would not call it a flaw.
![default](https://user-images.githubusercontent.com/1931904/51757328-23f62400-20d4-11e9-94b9-6829a1392bcf.png)

Reply:
But yes, in its current form it breaks item coloring in the item lists.

Reply:
> But yes, in its current form it breaks item coloring in the item lists.

To put it another way, I think `item::tname` should not dictate colour (beyond the damage indicator); leave it to the caller to colour as they wish to fit the context.  What makes sense in the inventory may not make sense elsewhere.  That doesn't work right now, but only because of a bug which #27856 hopefully fixes.

The means to provide a default colour for popups makes sense, though.

Reply:
You're right, it breaks too much. It's reverted now.